### PR TITLE
feat: add app setting page

### PR DESCRIPTION
**Pull request by windforce852**
Main feature:
add AppSetting page, which contains:
    'language' drop down, connected with a sample data array.
    'theme' drop down, with two staic options 'light', 'dark'
added entry from side nav bar

cp36-2
cp36-1

### CODE REVIEW

File: src/routes/routeConfig.ts [hunk 1]
```diff
@@ -38,6 +39,7 @@ export const ROUTE_KEY = {
   CRYPTO_PRICES: 'crypto_prices',
   CRYPTO_EXCHANGE_DETAILS: 'crypto_exchange_detail',
   CRYPTO_PRICE_DETAILS: 'crypto_price_detail',
+  APPSETTING: 'app_setting',
```
Comment (Chris-Wang) [inline src/routes/routeConfig.ts]: naming not aligned with others, use `APP_SETTING`

File: src/routes/routeConfig.ts [hunk 2]
```diff
@@ -140,6 +142,12 @@ export const ROUTE_CONFIG: Record<string, IRoute> = {
     title: 'Settings - BeeQuant',
     component: Settings,
   },
+  [ROUTE_KEY.APPSETTING]: {
+    path: '/app-setting',
```
Comment (Chris-Wang) [inline src/routes/routeConfig.ts]: we don't use `-` in path

File: tsconfig.json [hunk 3]
```diff
@@ -4,6 +4,7 @@
     "useDefineForClassFields": true,
     "lib": ["ES2020", "DOM", "DOM.Iterable"],
     "module": "ESNext",
+
```
Comment (Chris-Wang) [inline tsconfig.json]: ?

File: src/containers/AppSetting/AppSetting.tsx [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+  };
+
+  //Testing array, to be removed after implement the real language switching function
+  const languagesList: String[] = ['English', 'Chinese', 'Japanese', 'Korean'];
+
+  useTitle(getRouteByKey(ROUTE_KEY.APPSETTING).title);
+
+  return (
+    <Container>
+      {/*
```
Comment (Chris-Wang) [inline src/containers/AppSetting/AppSetting.tsx]: any pseudo code shouldn't be commented or added in pr

File: src/containers/AppSetting/AppSetting.tsx [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+} from '@/shared/components/form/FormElements';
+import { Controller, useForm } from 'react-hook-form';
+
+import { useUserContext } from '@/hooks/userHooks';
+import { THEME } from '@/shared/constants/storage';
+
+const AppSetting = () => {
+  const {
+    control,
+    // formState: { errors },
```
Comment (Chris-Wang) [inline src/containers/AppSetting/AppSetting.tsx]: commented code should be removed in pr

File: src/containers/AppSetting/AppSetting.tsx [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+              </CardTitleWrap>
+              <FormContainer $horizontal>
+                <FormGroup>
+                  <FormGroupLabel>Language</FormGroupLabel>
+                  <FormGroupField>
+                    <Controller
+                      name="Language"
+                      control={control}
+                      rules={{ required: 'Theme selection is required' }}
+                      defaultValue={'English'}
```
Comment (Chris-Wang) [inline src/containers/AppSetting/AppSetting.tsx]: Curly braces are unnecessary here.

File: src/containers/AppSetting/AppSetting.test.tsx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+Object.defineProperty(window, 'localStorage', {
+  value: mockLocalStorage,
+});
+
+// Mocking useUserContext
+jest.mock('@/hooks/userHooks', () => ({
+  useUserContext: jest.fn(),
+}));
+
+//Test case start
```
Comment (Chris-Wang) [inline src/containers/AppSetting/AppSetting.test.tsx]: space or tab after '//' in comment

File: src/containers/AppSetting/AppSetting.test.tsx [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+  });
+
+  it('should successfully render the page and show related texts', async () => {
+    render(
+      <Router>
+        <AppSetting />
+      </Router>
+    );
+
+    //Title should be rendered
```
Comment (Chris-Wang) [inline src/containers/AppSetting/AppSetting.test.tsx]: space or tab after '//' in comment

File: src/containers/AppSetting/AppSetting.test.tsx [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+    expect(screen.getByText('App Settings')).toBeInTheDocument();
+    expect(screen.getByText('Change your app settings')).toBeInTheDocument();
+
+    //Language drop down label should be rendered
```
Comment (Chris-Wang) [inline src/containers/AppSetting/AppSetting.test.tsx]: space or tab after '//' in comment

File: src/containers/AppSetting/AppSetting.test.tsx [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+    expect(screen.getByText('Language')).toBeInTheDocument();
+
+    //Theme drop down lable should be rendered
```
Comment (Chris-Wang) [inline src/containers/AppSetting/AppSetting.test.tsx]: space or tab after '//' in comment

File: src/containers/AppSetting/AppSetting.test.tsx [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+    expect(screen.getByText('Theme')).toBeInTheDocument();
+
+    //Theme drop down options should be rendered
```
Comment (Chris-Wang) [inline src/containers/AppSetting/AppSetting.test.tsx]: space or tab after '//' in comment

File: src/containers/AppSetting/AppSetting.test.tsx [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+
+// Mocking useUserContext
+jest.mock('@/hooks/userHooks', () => ({
+  useUserContext: jest.fn(),
+}));
+
+//Test case start
+describe('<AppSetting />', () => {
+  beforeAll(() => {
+    //simulates the current state managed by the context which represents the "live" state
```
Comment (Chris-Wang) [inline src/containers/AppSetting/AppSetting.test.tsx]: space or tab after '//' in comment

**Review by Chris-Wang — CHANGES_REQUESTED** [review]
Test failed:
image
Lack unit test:
- simulate user interactions and check if the state updates 
- assertions for the presence of language options in the dropdown
- edge case, e.g. local storage returns null or an unexpected value

File: src/containers/AppSetting/AppSetting.tsx (hunk 6 above)
Comment (windforce852) [reply src/containers/AppSetting/AppSetting.tsx]: compiler report bug after remove the inner curly braces
Screenshot 2024-04-19 at 7 02 32 AM

Comment (Chris-Wang) [reply src/containers/AppSetting/AppSetting.tsx]: Curly braces on ` defaultValue`

**Review by Chris-Wang — CHANGES_REQUESTED** [review]
Need to:
- add more unit test
- fix unit test error and pass pipeline
